presto-main/src/main/java/io/prestosql/metadata/MetadataListing.java

```diff
@@ -69,10 +70,23 @@ private MetadataListing() {}
 
     public static Set<SchemaTableName> listViews(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix)
     {
-        Set<SchemaTableName> tableNames = metadata.listViews(session, prefix).stream()
-                .map(QualifiedObjectName::asSchemaTableName)
-                .collect(toImmutableSet());
-        return accessControl.filterTables(session.toSecurityContext(), prefix.getCatalogName(), tableNames);
+        return getViews(session, metadata, accessControl, prefix).keySet();
```

Comment:
I don't think `listViews` should depend on `getViews`. Notice that you changed the `metadata.listViews(...)` to `metadata.getViews(...)`. These method could be implemented completely different (one could be cheap and other expensive to call). One only list objects, while the other retrieves all info about them.